main/config.c

```diff
@@ -2110,15 +2185,28 @@ static int process_text_line(struct ast_config *cfg, struct ast_category **cat,
 		}
 		/* A #include */
 		/* record this inclusion */
-		ast_include_new(cfg, cfg->include_level == 1 ? "" : configfile, cur, !do_include, cur2, lineno, real_inclusion_name, sizeof(real_inclusion_name));
+		newinclude = ast_include_new(cfg, cfg->include_level == 1 ? "" : configfile, cur, include_type, cur2, lineno, real_inclusion_name, sizeof(real_inclusion_name));
+		if (!newinclude) {
+			return -1;
+		}
 
-		do_include = ast_config_internal_load(cur, cfg, flags, real_inclusion_name, who_asked) ? 1 : 0;
+		result = ast_config_internal_load(cur, cfg, flags, real_inclusion_name, who_asked);
 		if (!ast_strlen_zero(exec_file))
 			unlink(exec_file);
-		if (!do_include && !try_include) {
+		if (!result && (include_type >= 0)) {
 			ast_log(LOG_ERROR, "The file '%s' was listed as a #include but it does not exist.\n", cur);
+			ast_include_destroy(newinclude);
 			return -1;
 		}
+
+		/* add comments */
+		if (ast_test_flag(&flags, CONFIG_FLAG_WITHCOMMENTS))
```

Comment:
these need braces

Reply:
I agree.  But, I was following the coding style already present in the file (in MANY places).

Reply:
Those are grandfathered in. New changes are not.

Reply:
Braces added.